holoviews/operation/datashader.py

```diff
@@ -1725,3 +1725,193 @@ class directly_connect_edges(_connect_edges, connect_edges):
+    Operation used to display the inspection mask, for use with other
+    inspection operations. Can be used directly but is more commonly
+    constructed using the mask property of the corresponding inspector
+    operation.
+    """
+
+    pixels = param.Integer(default=3, doc="""
+       Size of the mask that should match the pixels parameter used in
+       the associated inspection operation.""")
+
+    streams = param.List(default=[PointerXY])
+    x = param.Number(default=0)
+    y = param.Number(default=0)
+
+    @classmethod
+    def _distance_args(cls, element, x_range, y_range,  pixels):
+        ycount, xcount =  element.interface.shape(element, gridded=True)
+        x_delta = abs(x_range[1] - x_range[0]) / xcount
+        y_delta = abs(y_range[1] - y_range[0]) / ycount
+        return (x_delta*pixels, y_delta*pixels)
+
+    def _process(self, raster, key=None):
+        if isinstance(raster, RGB):
+            raster = raster[..., raster.vdims[-1]]
+        x_range, y_range = raster.range(0), raster.range(1)
+        xdelta, ydelta = self._distance_args(raster, x_range, y_range, self.p.pixels)
+        x, y = self.p.x, self.p.y
+        return self._indicator(raster.kdims, x, y, xdelta, ydelta)
+
+    def _indicator(self, kdims, x, y, xdelta, ydelta):
+        rect = np.array([(x-xdelta/2,y-ydelta/2), (x+xdelta/2, y-ydelta/2),
+                         (x+xdelta/2, y+ydelta/2), (x-xdelta/2, y+ydelta/2)])
+        data = {(str(kdims[0]),str(kdims[1])):rect}
+        return Polygons(data, kdims=kdims)
+
+
+class inspect_points(Operation):
+    """
+    Given datashaded aggregate (Image) output, return a single
+    (hoverable) point at the sample closest to the cursor.
+    """
+
+    pixels = param.Integer(default=3, doc="""
+       Number of pixels in data space around the cursor point to search
+       for hits in. The hit within this box mask that is closest to the
+       cursors position is displayed.""")
+
+    null_value = param.Number(default=0, doc="""
+       Value of raster which indicates no hits. For instance zero for
+       count aggregator (default) and commonly NaN for other (float)
+       aggregators. For RGBA images, the alpha channel is used which means
+       zero alpha acts as the null value.""")
+
+    value_bounds = param.NumericTuple(default=None, length=2, allow_None=True, doc="""
+       If not None, a numeric bounds for the pixel under the cursor in
+       order for hits to be computed. Useful for count aggregators where
+       a value of (1,1000) would make sure no more than a thousand
+       samples will be searched.""")
+
+    hits = param.DataFrame(default=pd.DataFrame(), allow_None=True)
+
+    point_count = param.Integer(default=1, doc="""
+       Maximum number of points to display within the mask of size
+       pixels. Points are prioritized by distance from the cursor
+       point. This means that the default value of one shows the single
+       closest sample to the cursor.""")
```

Comment:
Again, I'm not sure that's well defined; seems like all datapoints falling into the cursor's pixel are equivalent, and they don't need to be the closest (e.g. to the cursor pixel's center) to be appropriate; I'd just take any of them as there's no particular order that can be established on them.

Reply:
>  I'd just take any of them as there's no particular order that can be established on them.

That is a possible optimization and not true in all cases. 

If the datashading is occurring at screen resolution and bokeh's hover/tap is at the same resolution then it is true (well, the above is perfectly accurate but limited by the precision supplied by Bokeh), but it is very easy to ask `rasterize` or `datashade` to output rasters where that is not true at all (i.e bins much bigger than screen pixels by specifying custom `x_sampling` and `y_sampling`).

As currently defined, this inspector will work correctly for *all* datashaded output and this docstring is correct. For a lot of common datashader output where things are configured to automatically make datashaded bins match screen pixels, then your suggested change would work because Bokeh isn't giving anything past screen pixel precision (and even this might not be the case, for all I know a mouse pointer can specify an x,y position that is finer than a pixel! What happens if you stick your screen into 640x480 resolution?). 

Because of all this, I think the docstring is well defined and correct even if the precision with which the cursor's position is specified is limited.